test/node-fetch/main.js

```diff
@@ -0,0 +1,2310 @@
+  //     expect(res.headers.get('content-encoding')).to.equal('gzip')
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.equal('hello world')
+  //     })
+  //   })
+  // })
+
+  // it('should decompress deflate response', () => {
+  //   const url = `${base}deflate`
+  //   return fetch(url).then(res => {
+  //     expect(res.headers.get('content-type')).to.equal('text/plain')
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.equal('hello world')
+  //     })
+  //   })
+  // })
+
+  // it('should decompress deflate raw response from old apache server', () => {
+  //   const url = `${base}deflate-raw`
+  //   return fetch(url).then(res => {
+  //     expect(res.headers.get('content-type')).to.equal('text/plain')
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.equal('hello world')
+  //     })
+  //   })
+  // })
+
+  // it('should decompress brotli response', function () {
+  //   if (typeof zlib.createBrotliDecompress !== 'function') {
+  //     this.skip()
+  //   }
+
+  //   const url = `${base}brotli`
+  //   return fetch(url).then(res => {
+  //     expect(res.headers.get('content-type')).to.equal('text/plain')
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.equal('hello world')
+  //     })
+  //   })
+  // })
+
+  // it('should handle no content response with brotli encoding', function () {
+  //   if (typeof zlib.createBrotliDecompress !== 'function') {
+  //     this.skip()
+  //   }
+
+  //   const url = `${base}no-content/brotli`
+  //   return fetch(url).then(res => {
+  //     expect(res.status).to.equal(204)
+  //     expect(res.statusText).to.equal('No Content')
+  //     expect(res.headers.get('content-encoding')).to.equal('br')
+  //     expect(res.ok).to.be.true
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.be.empty
+  //     })
+  //   })
+  // })
+
+  // it('should skip decompression if unsupported', () => {
+  //   const url = `${base}sdch`
+  //   return fetch(url).then(res => {
+  //     expect(res.headers.get('content-type')).to.equal('text/plain')
+  //     return res.text().then(result => {
+  //       expect(result).to.be.a('string')
+  //       expect(result).to.equal('fake sdch string')
+  //     })
+  //   })
+  // })
+
+  // it('should reject if response compression is invalid', () => {
+  //   const url = `${base}invalid-content-encoding`
+  //   return fetch(url).then(res => {
+  //     expect(res.headers.get('content-type')).to.equal('text/plain')
+  //     return expect(res.text()).to.eventually.be.rejected
+  //       .and.be.an.instanceOf(FetchError)
+  //       .and.have.property('code', 'Z_DATA_ERROR')
+  //   })
+  // })
```

Comment:
Are these compression/descompression stuff part of spec or a node-fetch extension?

Reply:
I took a quick look and couldn't find anything in this regards in the spec. Looks like Google Chrome supports compression/decompression:

```
accept-encoding: gzip, deflate, br
```

Reply:
Firefox as well

Reply:
I guess we should open an issue in regards to whether or not we should support this if it's not in spec. Would you mind doing that and write your findings?